### PR TITLE
Gbs

### PR DESCRIPTION
**Pull request by AntoineRestivo**
- Added new benchmarks (Pcvl has still to be fixed)
- Updated documentation
- Circuit constructor

### CODE REVIEW

File: benchmarks/pcvl_data.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import quandelibc as qc
```
Comment (benoitseron) [inline benchmarks/pcvl_data.py:1]: Don't you think you need to leave the license as in their GitHub file ?

**Review by benoitseron — COMMENTED** [review]
Maybe the only thing would be to change the "::Float64" to Real, and maybe add a test to see if your circuit building reproduces the "analytical counter interferometer" of generalized bunching

File: benchmarks/pcvl_data.py (hunk 1 above)
Comment (AntoineRestivo) [reply benchmarks/pcvl_data.py:1]: Don't know what you are talking about

**Comment by AntoineRestivo** [comment]
> 

where in the package? I cannot find your comment in the review

File: benchmarks/pcvl_data.py (hunk 1 above)
Comment (benoitseron) [reply benchmarks/pcvl_data.py:1]: Up to line 23 https://github.com/Quandela/Perceval/blob/main/scripts/performance.py

File: src/types/interferometers.jl [hunk 2]
```diff
@@ -102,32 +102,71 @@ struct Rotation <: Interferometer
...
-    PhaseShift(shifted_modes::Array, param_::Array)
+    PhaseShift(phase::Float64)
 
-Creates a phase-shifter that is applied on the modes precised by shifted_modes with phase shifts given in param_.
+Creates a phase-shifter with parameter `phase`.
 
     Fields:
-        - shifted_modes::Array
-        - param_::Array
+        - phase::Float64
```
Comment (benoitseron) [inline src/types/interferometers.jl:110]: here for instance

File: src/types/interferometers.jl [hunk 3]
```diff
@@ -102,32 +102,71 @@ struct Rotation <: Interferometer
...
         - m::Int
         - U::Matrix{ComplexF64}
 """
 struct PhaseShift <: Interferometer
-    shifted_modes::Array
-    param_::Array
-    m::Int
+    phase::Float64
```
Comment (benoitseron) [inline src/types/interferometers.jl:115]: same

File: src/types/interferometers.jl [hunk 4]
```diff
@@ -102,32 +102,71 @@ struct Rotation <: Interferometer
...
+
+    @argcheck interf.m == length(target_modes)
+
+    push!(circuit.circuit_elements, interf)
+    circuit.U == nothing ? circuit.U = Matrix{ComplexF64}(I, circuit.m, circuit.m) : nothing
+
+    if interf.m == circuit.m
+        circuit.U = interf.U * circuit.U
+    else
+        u = Matrix{ComplexF64}(I, circuit.m, circuit.m)
```
Comment (benoitseron) [inline src/types/interferometers.jl:160]: this too

**Review by benoitseron — COMMENTED** [review]
pointed out some places for types Float etc, probably not all of them

**Pull request merged**